src/main/java/com/exactpro/th2/mstore/AbstractMessageStore.java

```diff
@@ -286,15 +281,45 @@ private void verifyBatch(T delivery) {
             if (previousKey == null) {
                 previousKey = sessionKey;
             }
-
+            if(!previousKey.equals(sessionKey)){
```

Comment:
This will prevent group batches to have different sessions and directions.
That is incorrect: batches contain different sessions and directions